### PR TITLE
0.3.0 socketio specify transport types

### PR DESCRIPTION
**Pull request by nshaaban-cPacket**
Depends on:
- https://github.com/1c3t3a/rust-socketio/pull/98

Socketio half of #94

### CODE REVIEW

File: socketio/src/client/socket.rs [hunk 1]
```diff
@@ -20,6 +22,16 @@ pub struct Socket {
 
 type SocketCallback = dyn FnMut(Payload, Socket) + 'static + Sync + Send;
 
+/// Flavor of engineio transport
+//TODO: Consider longevity
```
Comment (1c3t3a) [inline socketio/src/client/socket.rs:26]: What do you mean by that?

File: socketio/src/client/socket.rs [hunk 2]
```diff
@@ -192,13 +214,29 @@ impl SocketBuilder {
...
+        }
+
+        let mut builder = EngineIoSocketBuilder::new(url);
+
+        if let Some(tls_config) = self.tls_config {
+            builder = builder.tls_config(tls_config);
+        }
+        if let Some(headers) = self.opening_headers {
+            builder = builder.headers(headers);
+        }
```
Comment (1c3t3a) [inline socketio/src/client/socket.rs:230]: Would it make sense to have `builder.headers()` or `builder.tls_config()` taking an `Option<T>`? That way we wouldn't have the if let here and the builder holds them as `Option` anyways.

Comment (1c3t3a) [reply socketio/src/client/socket.rs:230]: But it feels strange to call a setter with an optional value...

Comment (nshaaban-cPacket) [reply socketio/src/client/socket.rs:230]: > But it feels strange to call a setter with an optional value...

Ultimately our primary goal is to provide an easy to use interface to the consumers of this API, if that means we need to have a tiny bit more boilerplate to save the users from writing Some() around every call to the setters headers and tls_config so be it. In the future this EngineIO builder will also be public.

File: socketio/src/client/socket.rs (hunk 1 above)
Comment (nshaaban-cPacket) [reply socketio/src/client/socket.rs:26]: Is storing the underlying transport type in an enum (with those options) what we want to use in the future? Moving or modifying (even adding) the enum in the future is a breaking change.

File: socketio/src/client/socket.rs (hunk 2 above)
Comment (1c3t3a) [reply socketio/src/client/socket.rs:230]: Yep that's true.

**Pull request merged**